Http/Controllers/API/V1/PriceController.php

```diff
@@ -47,33 +47,38 @@ public function index(Request $request)
     /**
      * Builds an array of List IDs from request parameters.
      *
-     * @param  \Illuminate\\Http\Request  $request
-     * @return array
+     * @param  \Illuminate\Http\Request  $request
+     * @return \App\Models\TenantPriceList[]
      */
     private function buildListsArrayFromRequestParams(Request $request){
 
         $lists = [];
 
         if ($request->has('list_id')) {
             $lists[] = $request->get('list_id');
-            return $lists;
         }
 
         if ($request->has('list_ids')) {
             $lists = [];
             foreach ( $request->array('list_ids') as $key => $listId ){
                 $lists[] = $listId;
             }
-            return $lists;
         }
 
         if ($request->has('reimbursement_type_id')) {
             $query = TenantPriceList::forReimbursementType($request->get('reimbursement_type_id'))
                 ->used();
             if ($priceList = $query->first()) {
-                $lists[] = $priceList->id;
+                $lists[] = $priceList;
+                return $lists;
             }
         }
+
+        if(empty($lists)){
+            return $lists;
+        }
+
+        $lists = TenantPriceList::whereIn(’id’,$lists);
```

Comment:
Main changes for this commit are in this method. At the end, with a single query execution $lists will be populated with hydrated models instead of the previous numerical values. In this way we have avoided the for loop and multiple query executions while achieving the same result.